index.js

```diff
@@ -83,7 +99,16 @@ function AtImport(options) {
  * @param {Object} styles
  * @param {Object} options
  */
-function parseStyles(styles, options, cb, importedFiles, ignoredAtRules, media, hashFiles) {
+function parseStyles(styles, options, cb, globallyImportedFiles, ignoredAtRules, media, hashFiles, cache) {
+  if (options.from && !media && cache && IsCached(options.from, options, cache)) {
```

Comment:
Media queried imports will be cached by the requesting file. One shouldn't cache the file when requested with a media query because it will alter the output from how it looks when requested normally.